### PR TITLE
[SYCL] correct sub-device count calculation for numa partitioning

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: sycl/source/detail/device_impl.cpp [hunk 1]
```diff
@@ -184,7 +192,32 @@ device_impl::create_sub_devices(const std::vector<size_t> &Counts) const {
...
-  Properties.insert(Properties.begin() + 1, Counts.begin(), Counts.end());
+
+  auto It = Properties.begin() + 1;
+  size_t TotalCounts = 0;
+  size_t NonZeroCounts = 0;
+  for (auto Count : Counts) {
+    TotalCounts += Count;
+    NonZeroCounts += (Count != 0) ? 1 : 0;
+    It = Properties.insert(It, Count);
+  }
```
Comment (steffenlarsen) [inline sycl/source/detail/device_impl.cpp:203]: Please add a comment about what is happening here.

**Review by steffenlarsen — COMMENTED** [review]
Only a small nit. Do we have tests for this?

File: sycl/source/detail/device_impl.cpp (hunk 1 above)
Comment (smaslov-intel) [reply sycl/source/detail/device_impl.cpp:204]: OK

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Pull request merged**